scripts/test/Muon/muon_file_utils_test.py

```diff
@@ -0,0 +1,66 @@
+import Muon.GUI.Common.muon_file_utils as utils
+import sys
+import os
+
+import unittest
+
+if sys.version_info.major == 3:
+    from unittest import mock
+else:
+    import mock
+
+
+class MuonFileUtilsTest(unittest.TestCase):
+
+    def test_parse_user_input_to_files_returns_single_file_as_list(self):
+        filename = "EMU0001234.nxs"
+        parsed_file = utils.parse_user_input_to_files(filename)
+
+        self.assertEqual(parsed_file, [filename])
+
+    def test_parse_user_input_to_files_returns_full_filepath(self):
+        files = ["C:" + os.sep + "dir1" + os.sep + "dir2" + os.sep + "EMU0001234.nxs",
+                 "dir1" + os.sep + "dir2" + os.sep + "EMU0001234.nxs"]
+
+        for file_name in files:
+            parsed_file = utils.parse_user_input_to_files(file_name)
+            self.assertEqual(parsed_file, [file_name])
```

Comment:
I thought parsed_file would be just the name: e.g. `EMU0001234.nxs`. So why does `[file_name]` (I think this should be the full path) match it?

Reply:
So if there is a directory (i.e. base) attached to the file it will be removed; in the case you mention there is no base and so nothing is removed, leaving the string unchanged.

Reply:
I thought the base would be `["C:" + os.sep + "dir1" + os.sep + "dir2" + os.sep `

Reply:
So the test is called `test_parse_user_input_to_files_returns_full_filepath`, so I'm testing that if I give the parser a full filepath, that it just returns the full filepath. This would contrast to cases where there were multiple filepaths, and then the output of the parser would be different from the input.

Because this behaviour isn't necessarily obvious, I have a unit test with this name to make clear that the function returns full filepaths.